### PR TITLE
[TACHYON-1116] Making worker count configurable

### PR DESCRIPTION
**Pull request by aslanbekirov**
Making worker count configurable for tachyon on mesos

### CODE REVIEW

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java [hunk 1]
```diff
@@ -140,11 +142,12 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
           targetMem = masterMem;
           mMasterHostname = offer.getHostname();
           mTaskName = sConf.get(Constants.INTEGRATION_MESOS_TACHYON_MASTER_NAME);
-          mMasterCount ++;
+          mMasterCount++;
```
Comment (calvinjia) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: Minor style: `mMasterCount ++` (Applies to other areas as well)

**Comment by uronce-cc** [comment]
+1 LGTM

File: common/src/main/resources/tachyon-default.properties [hunk 2]
```diff
@@ -42,6 +42,10 @@ tachyon.zookeeper.leader.path=/leader
 tachyon.integration.mesos.executor.dependency.path=https://s3.amazonaws.com/tachyon-mesos
 tachyon.integration.mesos.jre.url=https://s3.amazonaws.com/tachyon-mesos/jre-7u79-linux-x64.tar.gz
 tachyon.integration.mesos.jre.version=jre1.7.0_79
+tachyon.integration.mesos.master.node.count=1
+tachyon.integration.mesos.worker.node.count=1
+tachyon.integration.mesos.master.name=masternode
+tachyon.integration.mesos.worker.name=workernode
```
Comment (apc999) [inline common/src/main/resources/tachyon-default.properties]: could you keep these properties in this section still in alphabetical order?

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java [hunk 3]
```diff
@@ -242,6 +247,8 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
         case TASK_ERROR:
           if (status.getTaskId().getValue().equals(String.valueOf(mMasterTaskId))) {
             mMasterCount --;
+          } else if (status.getTaskId().getValue().equals(String.valueOf(mWorkerTaskId))) {
+            mWorkerCount --;
```
Comment (apc999) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: what does this mean? A worker is lost? We'd better print something. Same as the case of `mMasterCount --`

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java [hunk 4]
```diff
@@ -48,9 +48,11 @@
     private String mMasterHostname = "";
     private String mTaskName = "";
     private int mMasterTaskId;
+    private int mWorkerTaskId;
```
Comment (apc999) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: what does `mWorkerTaskId` mean here? Maybe my understanding is wrong, aren't we expected to have multiple worker tasks? Shouldn't they have different Ids?

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java [hunk 5]
```diff
@@ -242,6 +247,8 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
         case TASK_ERROR:
           if (status.getTaskId().getValue().equals(String.valueOf(mMasterTaskId))) {
             mMasterCount --;
+          } else if (status.getTaskId().getValue().equals(String.valueOf(mWorkerTaskId))) {
```
Comment (apc999) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: what if the status corresponds to a previous worker task?

**Comment by apc999** [comment]
@aslanbekirov , I left a bunch questions---likely because I didn't understand well the mesos workflow. Thanks for the updates.

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java (hunk 3 above)
Comment (aslanbekirov) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: This means that a worker or master task is lost or failed. I will add some message

File: integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java (hunk 5 above)
Comment (jsimsa) [inline integration/mesos/src/main/java/tachyon/mesos/TachyonFramework.java]: @apc99 is right ... you cannot use mWorkerTaskId to identify all possible worker

instead you should be comparing task name against `Constants.INTEGRATION_MESOS_TACHYON_MASTER_NAME` and `Constants.INTEGRATION_MESOS_TACHYON_WORKER_NAME`

At that point, you can get rid of `mMasterTaskId` and `mWorkerTaskId`.

**Comment by jsimsa** [comment]
@aslanbekirov thanks for the PR, left you some comments ... also please resolve merge conflicts

**Comment by aslanbekirov** [comment]
Hi @jsimsa and @apc999 , really sorry, I was in vacation and out of internet, I am gonna be back in few days. I will resolve and continue PR. Thanks..

**Comment by jsimsa** [comment]
@aslanbekirov how is it going? could we please wrap up this PR?

**Comment by aslanbekirov** [comment]
I think we can close this PR and I will create new PR for this in 2 days.

**Comment by jsimsa** [comment]
Sounds good.

**Pull request closed without merging**